pulsar-common/src/main/java/org/apache/pulsar/common/api/PulsarDecoder.java

```diff
@@ -65,11 +69,30 @@
 import org.slf4j.LoggerFactory;
 
 public abstract class PulsarDecoder extends ChannelInboundHandlerAdapter {
-
     // Max message size is limited by max BookKeeper entry size which is 5MB, and we need to account
     // for headers as well.
-    public final static int MaxMessageSize = (5 * 1024 * 1024 - (10 * 1024));
-    public final static int MaxFrameSize = 5 * 1024 * 1024;
+    public static int MaxMessageSize;
+    public static int MaxFrameSize;
+
+    private static String commonConfigFile = Paths.get("").toAbsolutePath().normalize().toString() + "/conf/common.conf";
```

Comment:
how can a client load this setting? 

I am not sure it is a good approach to load such settings from a config file. A system property or environment variable (e.g. org.apache.pulsar.MaxMessageSize = xyz) is better.